### ISSUE TITLE
[a11y]: [UI Shell] [Header with Actions & Side Nav] - Category title buttons lack keyboard focus during tab navigation

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16886]
### Package

@carbon/react

### Browser

Firefox

### Operating System

Windows

### Package version

v1.58.0

### React version

v1.60.3

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

<img width="418" alt="image" src="https://github.com/carbon-design-system/carbon/assets/63502271/4aedde1b-d50e-4532-82b6-10a7b4552447">

When navigating through the tab key, the category title buttons are not receiving keyboard focus. This prevents users who rely on keyboard navigation from accessing these buttons

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-ui-shell-header--header-w-navigation-actions-and-side-nav

### Steps to reproduce

https://react.carbondesignsystem.com/?path=/story/components-ui-shell-header--header-w-navigation-actions-and-side-nav
Visit the storybook via above link
Navigate the webpage via keyboard (Tab key) and observe that the focus does not goes to category title buttons but directly moves to the Link button below them

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
